pkg/inject/sidecar_test.go

```diff
@@ -474,3 +498,42 @@ func containsOptionWithPrefix(t *testing.T, args []string, prefix string) bool {
 	assert.Fail(t, "list of arguments didn't have an option starting with '%s'", prefix)
 	return false
 }
+
+func containsAgentTag(t *testing.T, args []string, tag string, value string) bool {
+	prefix := "--jaeger.tags="
+	expectedTagValue := fmt.Sprintf("%s=%s", tag, value)
+
+	for _, arg := range args {
+		if strings.HasPrefix(arg, prefix) {
+			tagParam := strings.SplitN(arg, "=", 2)
+			for _, tagValue := range strings.Split(tagParam[1], ",") {
+				if tagValue == expectedTagValue {
+					return true
+				}
+			}
+			assert.Fail(t, "agent tag '%s=%s' not found", tag, value)
+			return false
+		}
+	}
+	assert.Fail(t, "list of arguments didn't have an option starting with '%s'", prefix)
+	return false
+}
+
+func doesNotContainAgentTag(t *testing.T, args []string, tag string, value string) bool {
+	prefix := "--jaeger.tags="
```

Comment:
Won't the `t.Fail` calls then occur in the "pass" flow?

Reply:
True. Instead of having the assert.Fail inside the `containsAgent...` function, you can keep the assertion only at the top-level functions, like:

```golang
func TestBla(t *testing.T)  {
  assert.True(t, containsAgentTag(...))
}
```